### PR TITLE
feat: superadmin delete organisation by id

### PR DESCRIPTION
**Pull request by chionz**
## Description

This pull request introduces a new endpoint that allows superadmin users to delete an organization by its ID. The implementation ensures that only authorized superadmin accounts can perform this action.

Key Changes:
- Created a DELETE endpoint at `/api/v1/superadmin/organizations/{org_id}`.
- Implemented authorization checks to ensure only superadmin users can access this endpoint.
- Added logic to handle cases where the specified organization does not exist, returning a 404 Not Found status.
- Implemented error handling to return a 500 Internal Server Error in case of unexpected issues.

## Related Issue (Link to issue ticket)

issues 371 https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/371

## Motivation and Context

This change is required to provide superadmin users with the capability to delete organizations within the application. It enhances security by restricting this functionality to authorized users and ensures proper error handling and response statuses, aligning with the defined acceptance criteria.

## How Has This Been Tested?

The implementation has been tested as follows:
- **Successful Deletion**: Verified that an organization is successfully deleted when a valid ID is provided by a superadmin.
- **Organization Not Found**: Ensured that a 404 Not Found response is returned when attempting to delete a non-existent organization.
- **Unauthorized Access**: Confirmed that non-superadmin users are unable to access the delete endpoint.
- **Server Error Handling**: Simulated server errors to verify that a 500 Internal Server Error response is appropriately returned.

Tests were conducted using [appropriate testing tools/environments] to validate the functionality and ensure no regression in the codebase.

## Screenshots (if appropriate - Postman, etc):

Screenshot from 2024-07-30 04-09-35

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/organization.py [hunk 1]
```diff
@@ -55,12 +57,25 @@ async def update_organization(
...
         data = jsonable_encoder(orgs)
-    )
\ No newline at end of file
+    )
+
+@organization.delete("/{org_id}")
+async def delete_organization(org_id: str, db: Session = Depends(get_db), 
+                        current_user: User = Depends(user_service.get_current_super_admin),
+):
+    organization = db.query(Organization).filter(Organization.id == org_id).first()
```
Comment (joboy-dev) [inline api/v1/routes/organization.py:73]: Don't make db queries in the routes. Do it with services.

File: api/v1/services/organization.py
Comment (joboy-dev) [inline api/v1/services/organization.py:1]: Can you update this function
There's a lot of typos so instead of product, it should be organization 

Thanks

**Review by joboy-dev — APPROVED** [review]

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

**Comment by chionz** [comment]
@joboy-dev I have made changes.
please check

**Review by joboy-dev — APPROVED** [review]

**Review by joboy-dev — APPROVED** [review]

**Review by Goketech — APPROVED** [review]

**Pull request merged**